cpu/native/mtd/Kconfig

```diff
@@ -5,8 +5,8 @@
 # directory for more details.
 #
 
-config MODULE_MTD_NATIVE
+config HAVE_MTD_NATIVE
```

Comment:
We could group all these MTD feature symbols under the single mtd kconfig file, as you're doing with the modules. Also, now that we are at it, it could make sense to add the symbols for the other backends.

Reply:
Sure, other than that does the re-strucure make sense?

Reply:
> Sure, other than that does the re-strucure make sense?

I mean the dependency re-structuring done in this PR.

Reply:
> > Sure, other than that does the re-strucure make sense?
> 
> I mean the dependency re-structuring done in this PR.

I think it makes sense in general, but we want board to define their default backend, so we need to add features indicating this for the rest of them (maybe for the ones depending on drivers having the driver is enough?).

Also, as now this enforces no interface for MTD via a dependency, we should catch this condition with some error symbol, that would stop compilation without an interface:

```
comment "Please select an interface for MTD"
    depends on MTD_NO_INTERFACE

config ERROR_MTD_NO_INTERFACE
    string
    default "An interface for MTD should be selected"
    depends on MTD_NO_INTERFACE

config MTD_NO_INTERFACE
    bool
    default y
    depends on !MODULE_MTD_SPI_NOR && !MODULE_MTD_FLASHPAGE && !MODULE_MTD_NATIVE
    depends on !MODULE_MTD_AT24CXXX && !MODULE_MTD_AT25XXX && !MODULE_MTD_MCI && !MODULE_MTD_SDCARD
    help
        Checks if a valid MTD interface has been set.
```

Reply:
I disagree, this is not done currently because an MTD interface can be mocked easily with some RAM buffer, is it used this way in the tests, as stated in the commit message 4f9d0660740f718d6983b61229d3f7632f3626ed. We could add an explicit mock for this but IMO this is out of scope.

Reply:
Ah, I missed that part. If MTD is available without interfaces then no need for this. Sorry about the noise.

Reply:
I mean you are not wrong, it's just that currently, nothing forbids including the API with no backend ( some tests just exit, or say there is no backend available without an error), in other tests tests it's mocked this way https://github.com/RIOT-OS/RIOT/blob/ffff68deaff4f204da38a7d2c9f902efcdf22261/tests/unittests/tests-mtd/tests-mtd.c#L28-L96.

Reply:
Ok, then for now we leave it like this.